src/networks/infinitesquarenetwork.jl

```diff
@@ -1,67 +1,84 @@
-Base.eltype(::Type{<:InfiniteSquareNetwork{T}}) where {T} = T
-Base.eltype(A::InfiniteSquareNetwork) = eltype(typeof(A))
+unitcell(n::InfiniteSquareNetwork) = n.A
+Base.size(n::InfiniteSquareNetwork, args...) = size(unitcell(n), args...)
+Base.length(n::InfiniteSquareNetwork) = length(unitcell(n))
+Base.eltype(n::InfiniteSquareNetwork) = eltype(typeof(n))
+Base.eltype(::Type{InfiniteSquareNetwork{O}}) where {O} = O
 
-## Copy
-Base.copy(A::NWType) where {NWType<:InfiniteSquareNetwork} = NWType(copy(unitcell(A)))
-function Base.similar(A::NWType, args...) where {NWType<:InfiniteSquareNetwork}
-    return NWType(similar(unitcell(A), args...))
+Base.copy(n::InfiniteSquareNetwork) = InfiniteSquareNetwork(copy(unitcell(n)))
+function Base.similar(n::InfiniteSquareNetwork, args...)
+    return InfiniteSquareNetwork(similar(unitcell(n), args...))
 end
-function Base.repeat(A::NWType, counts...) where {NWType<:InfiniteSquareNetwork}
-    return NWType(repeat(unitcell(A), counts...))
+function Base.repeat(n::InfiniteSquareNetwork, counts...)
+    return InfiniteSquareNetwork(repeat(unitcell(n), counts...))
 end
 
 ## Indexing
-Base.getindex(A::InfiniteSquareNetwork, args...) = Base.getindex(unitcell(A), args...)
-function Base.setindex!(A::InfiniteSquareNetwork, args...)
-    return (Base.setindex!(unitcell(A), args...); A)
+Base.getindex(n::InfiniteSquareNetwork, args...) = Base.getindex(unitcell(n), args...)
+function Base.setindex!(n::InfiniteSquareNetwork, args...)
+    return (Base.setindex!(unitcell(n), args...); n)
 end
-Base.axes(A::InfiniteSquareNetwork, args...) = axes(unitcell(A), args...)
-function eachcoordinate(A::InfiniteSquareNetwork)
-    return collect(Iterators.product(axes(A)...))
-end
-function eachcoordinate(A::InfiniteSquareNetwork, dirs)
-    return collect(Iterators.product(dirs, axes(A, 1), axes(A, 2)))
+Base.axes(n::InfiniteSquareNetwork, args...) = axes(unitcell(n), args...)
+eachcoordinate(n::InfiniteSquareNetwork) = collect(Iterators.product(axes(n)...))
+function eachcoordinate(n::InfiniteSquareNetwork, dirs)
+    return collect(Iterators.product(dirs, axes(n, 1), axes(n, 2)))
 end
 
+## Spaces
+
+virtualspace(n::InfiniteSquareNetwork, r::Int, c::Int, dir) = virtualspace(n[r, c], dir)
+
 ## Vector interface
-function VectorInterface.scalartype(::Type{NWType}) where {NWType<:InfiniteSquareNetwork}
-    return scalartype(eltype(NWType))
-end
-function VectorInterface.zerovector(A::NWType) where {NWType<:InfiniteSquareNetwork}
-    return NWType(zerovector(unitcell(A)))
+VectorInterface.scalartype(::Type{<:InfiniteSquareNetwork{O}}) where {O} = scalartype(O)
+function VectorInterface.zerovector(A::InfiniteSquareNetwork)
+    return InfiniteSquareNetwork(zerovector(unitcell(A)))
 end
 
-## Math
+## Math (for Zygote accumulation)
+
 function Base.:+(A₁::NWType, A₂::NWType) where {NWType<:InfiniteSquareNetwork}
-    return NWType(unitcell(A₁) + unitcell(A₂))
+    return NWType(_add_localsandwich.(unitcell(A₁), unitcell(A₂)))
 end
```

Comment:
This signature is a bit strange because it would only allow things that have the exact same type (for example different scalartypes would fail. Maybe better:
```julia
Base.:+(A1::InfiniteSquareNetwork, A2::InfiniteSquareNetwork) = InfiniteSquareNetwork(_add_localsandwich.(unitcell(A1), unitcell(A2)))
```

I would also do the same for the methods below